pipeline/measuring.py

```diff
@@ -0,0 +1,307 @@
+            'adjust the annulus size for each image based on the PSF width. '
+        )
+
+        # TODO: should we choose the "best aperture" using the config, or should each Image have its own aperture?
+        self.chosen_aperture = self.add_par(
+            'chosen_aperture',
+            0,
+            [str, int],
+            'The aperture radius that is used for photometry. '
+            'Choose either the index in the aperture_radii list, '
+            'the string "psf", or the string "auto" to choose '
+            'the best aperture in each image separately. '
+        )
+
+        self.analytical_cuts = self.add_par(
+            'analytical_cuts',
+            ['negatives', 'bad pixels', 'offsets', 'filter bank'],
+            [list],
+            'Which kinds of analytic cuts are used to give scores to this measurement. '
+        )
+
+        self.negative_positive_threshold = self.add_par(
+            'negative_positive_threshold',
+            3.0,
+            float,
+            'How many times the local background RMS for each pixel counts '
+            'as being a negative or positive pixel. '
+        )
+
+        self.bad_pixel_radius = self.add_par(
+            'bad_pixel_radius',
+            3.0,
+            float,
+            'Radius in pixels for the bad pixel cut. '
+        )
+
+        self.bad_pixel_exclude = self.add_par(
+            'bad_pixel_exclude',
+            [],
+            list,
+            'List of strings of the bad pixel types to exclude from the bad pixel cut. '
+            'The same types are ignored when running photometry. '
+        )
+
+        self.streak_filter_angle_step = self.add_par(
+            'streak_filter_angle_step',
+            5.0,
+            float,
+            'Step in degrees for the streaks filter bank. '
+        )
+
+        self.width_filter_multipliers = self.add_par(
+            'width_filter_multipliers',
+            [0.25, 2.0, 5.0, 10.0],
+            list,
+            'Multipliers of the PSF width to use as matched filter templates'
+            'to compare against the real width (x1.0) when running psf width filter. '
+        )
+
+        self._enforce_no_new_attrs = True
+
+        self.override(kwargs)
+
+    def get_process_name(self):
+        return 'measuring'
+
+
+class Measurer:
+    def __init__(self, **kwargs):
+        self.pars = ParsMeasurer(**kwargs)
+
+        # this is useful for tests, where we can know if
+        # the object did any work or just loaded from DB or datastore
+        self.has_recalculated = False
+
+        self._filter_bank = None  # store an array of filters to match against the cutouts
+        self._filter_psf_fwhm = None  # recall the FWHM used to produce this filter bank, recalculate if it changes
+
+    def run(self, *args, **kwargs):
+        """
+        Go over the cutouts from an image and measure all sorts of things
+        for each cutout: photometry (flux, centroids), real/bogus, etc.
```

Comment:
real/bogus doesn't belong in this class; it should be done externally, as it's really a separate process.

(The analytical things like the elongation etc. do make sense to have here, as they really are properties of the photometry itself.  r/b is a property of photometry plus decisions about training, etc.  And, we want to have the abillity to add a new r/b method at any point without having to redo all photometry.)

Reply:
Yes, you are right, we need to add another model and processing step for R/B.